### PR TITLE
Code in `rti.c` made available as library

### PR DESCRIPTION
**Pull request by Jakio815**

### CODE REVIEW

**Comment by Jakio815** [comment]
I think it will work if the main branch reverts to commit `4f78b42a` which was the commit just before I started pushing. I also merged #169 together what you merged right after my PR. Would you please have a look? @lhstrh.

**Review by edwardalee — CHANGES_REQUESTED** [review]
This PR also seems to loose the history. Note that rti_lib.c is entirely authored by @Jakio815 even though it is mostly copied from rti.c.

**Comment by lhstrh** [comment]
The trick is to use `git mv` for the file that will carry most of the history and create new ones for the smaller ones... Whatever gets factored out, however, will not of course loose history, there is no way around that.

**Comment by lhstrh** [comment]
@Jakio815 did you see the link that @erlingrj shared about doing this?

**Comment by edwardalee** [comment]
You can preserve the history even on the smaller file, although it may not be worth the trouble if it is small enough. See:

https://devblogs.microsoft.com/oldnewthing/20190916-00/?p=102892

**Comment by Jakio815** [comment]
> 

I'm not sure what you are mentioning. Are you mentioning [this](https://devblogs.microsoft.com/oldnewthing/20190916-00/?p=102892)? 

And actually when I checkout to this branch, I see the file history. I followed the link above which professor @edwardalee mentioned. I see that the Git's Files Changed tap seems to change everything, but in my VSCode GitLens extension, it has the original history. I'm not sure if this is the right way. Would you please check? @lhstrh

**Comment by edwardalee** [comment]
Oh, interesting. Yes, it could be just that the history doesn't show up in Files Changed. I hadn't realized that, but it makes sense.

**Comment by Jakio815** [comment]
I also see that the contributors for rti_lib.c is only me. I'm not sure how I can check this on the Github web.

**Comment by Jakio815** [comment]
I found out that if you tap ['Blame'](https://github.com/lf-lang/reactor-c/blame/0e60d4b72811c011b5486f6d16278c8f9553bcb7/core/federated/RTI/rti_lib.c), you could see the history of every lines. Also, if you tap the ['History'](https://github.com/lf-lang/reactor-c/commits/84845c5391dc3f9c1f5d44cd832654d9bed3b99d/core/federated/RTI/rti_lib.c) you could check the history below saying that it was renamed.

Sorry for bothering everyone with this problem.

**Comment by lhstrh** [comment]
I reverted my merge commit of the previous PR, so once approved, you can just go ahead and hit the merge button. Please mark it as "ready for review" so that @edwardalee  can approve it. Thanks!

File: .github/workflows/ci.yml
Comment (edwardalee) [inline .github/workflows/ci.yml:23]: I guess this should change back to 
```
uses: lf-lang/reactor-c/.github/workflows/build-rti.yml@main
```
as part of the merge into main?

**Review by edwardalee — APPROVED** [review]
LGTM!  Thanks for fixing this.

File: .github/workflows/ci.yml
Comment (Jakio815) [reply .github/workflows/ci.yml:23]: Marten usually does it after merged to main.

Comment (Jakio815) [reply .github/workflows/ci.yml]: I just done it as the tests all passed.

**Pull request merged**
